.github/workflows/repo-metrics-weekly.yml

```diff
@@ -0,0 +1,58 @@
+name: Weekly repo metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * MON'
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  build:
+    # this workflow will always fail in forks; bail if this isn't running in the upstream
+    if: github.repository == 'aws/aws-cdk'
+    name: metrics
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Get dates for last week
+      shell: bash
+      run: |
+        # Calculate the previous Monday's date
+        PREVIOUS_MONDAY=$(date -d "last monday" "+%Y-%m-%d")
+
+        # Calculate the current Sunday's date
+        CURRENT_SUNDAY=$(date -d "next Sunday" "+%Y-%m-%d")
+
+        # Set an environment variable with the date range
+        echo "$PREVIOUS_MONDAY..$CURRENT_SUNDAY"
+        echo "last_week=$first_day..$last_day" >> "$GITHUB_ENV"
```

Comment:
Seems the variable needs to be updated with previous monday and current sunday

Reply:
Good catch!